python/ray/ml/trainer.py

```diff
@@ -133,6 +137,8 @@ def training_loop(self):
         resume_from_checkpoint: A checkpoint to resume training from.
     """
 
+    _scaling_config_allowed_keys = []
```

Comment:
```suggestion
    _scaling_config_allowed_keys: List[str]
```
We can actually define it like that, which would cause an exception to be raised on init if it's not overridden in a subclass

Reply:
If we do this, we should also consider turning `Trainer` into an abstract base class. Any reason why we didn't do this, yet? cc @matthewdeng @amogkam

Reply:
It currently is right?

Reply:
Then let's do it @Yard1

Reply:
Ah, quick question, I've just tried:

```
import abc
from typing import List


class Base(abc.ABC):
    _wurst: List[str]


Base()
```

which works fine, so this isn't actually validated from the base class but it fails on attribute access. If that's the case, let's go back to `[]` again, as we don't want an attribute access error as the default case.

Reply:
Thanks for checking, must have been misremembering the behavior. Let me revert